### PR TITLE
Fixes #15130 - use puppet apiv2 when 'environmentpath' is present.

### PR DESCRIPTION
**Pull request by dmitri-d**

### CODE REVIEW

File: test/puppet/puppet_legacy_configuration_test.rb [hunk 1]
```diff
@@ -90,14 +90,14 @@ def test_load_programmable_settings_sets_api_v2_for_environments_retriever_when_
     assert_equal :api_v2, configuration.load_programmable_settings(:puppet_version => "3.2", :use_environment_api => true)[:environments_retriever]
   end
 
-  def test_load_programmable_settings_sets_api_v2_for_environments_retriever_whithout_puppet_use_environment_api_and_environmentpath_missing
+  def test_load_programmable_settings_sets_config_file_for_environments_retriever_whithout_puppet_use_environment_api_and_environmentpath_missing
```
Comment (domcleal) [inline test/puppet/puppet_legacy_configuration_test.rb]: Nitpick, "whithout" => "without"

File: test/puppet/puppet_legacy_configuration_test.rb [hunk 2]
```diff
@@ -90,14 +90,14 @@ def test_load_programmable_settings_sets_api_v2_for_environments_retriever_when_
...
     configuration = LegacyProviderConfigurationForTesting.new(:master => {}, :main => {})
-    assert_equal :api_v2, configuration.load_programmable_settings(:puppet_version => "3.2")[:environments_retriever]
+    assert_equal :config_file, configuration.load_programmable_settings(:puppet_version => "3.2")[:environments_retriever]
   end
 
   def test_load_programmable_settings_sets_api_v2_for_environments_retriever_whithout_puppet_use_environment_api_and_environmentpath_present
```
Comment (domcleal) [inline test/puppet/puppet_legacy_configuration_test.rb]: ditto, speeling

File: test/puppet/puppet_legacy_configuration_test.rb [hunk 3]
```diff
@@ -155,7 +155,7 @@ def test_puppet_initializer_wiring
     assert @container.get_dependency(:puppet_initializer).instance_of?(Proxy::PuppetLegacy::Initializer)
   end
 
-  def test_cached_future_parser_wiring_parameters
+  def test_cached_future_parser_wiring_paramenvironmentpatheters
```
Comment (domcleal) [inline test/puppet/puppet_legacy_configuration_test.rb]: typo?

**Comment by dmitri-d** [comment]
rebased, fixed misspeelings.

**Comment by domcleal** [comment]
Merged as 185cfbb, thanks @witlessbird.

**Pull request closed without merging**